youtube_dl/extractor/bigo.py

```diff
@@ -0,0 +1,57 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import ExtractorError, urlencode_postdata
+
+
+class BigoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?bigo\.tv/(?:[a-z]{2,}/)?(?P<id>[^/]+)'
+
+    _TESTS = [{
+        'url': 'https://www.bigo.tv/ja/221338632',
+        'info_dict': {
+            'id': '6576287577575737440',
+            'title': '土よ〜💁‍♂️ 休憩室/REST room',
+            'thumbnail': r're:https?://.+',
+            'uploader': '✨Shin💫',
+            'uploader_id': '221338632',
+            'is_live': True,
+        },
+        'skip': 'livestream',
+    }, {
+        'url': 'https://www.bigo.tv/th/Tarlerm1304',
+        'only_matching': True,
+    }, {
+        'url': 'https://bigo.tv/115976881',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        user_id = self._match_id(url)
+
+        info_raw = self._download_json(
+            'https://bigo.tv/studio/getInternalStudioInfo',
+            user_id, data=urlencode_postdata({'siteId': user_id}))
+
+        if info_raw.get('code'):
```

Comment:
As there's an error check, be aware that the returned JSON object may not be a `dict`, so maybe `not isinstance(info_raw, dict) or ...`, or previously:
```
        if not isinstance(info_raw, dict):
            info_raw = {'code': '', 'msg': 'No valid JSON data', }
```
I'm sure you can think of a suitable solution.

Reply:
Once that happen, it means they changed how to call the API. Why not make it bail out?

Reply:
Let's hope that doesn't happen, then, but in case it does a specific log message, rather than "... object has no attribute 'get'", would surely be more helpful. Up to you, anyway.